design/open/commit-metadata/airflow.md

```diff
@@ -0,0 +1,178 @@
+# Metadata for Airflow integration
+
+## About
+
+This is a design to integrate metadata from Airflow into lakeFS.  It
+consists of metadata integration and UI for using it.
+
+## Metadata integration
+
+Metadata that Airflow operators and/or hooks may place on commits to
+associate them with the generating Airflow.
+
+### Stability
+
+Commit _metadata_, like diamonds, are forever: It is not generally possible to
+modify commit metadata.  Even if we were willing to rewrite commits and
+could do so safely, it would still break commit digests as stable
+identifiers.
+
+### Forwards compatibility
+
+Obviously we cannot guarantee any future metadata compatibility.  However
+_where current implementation cost is unaffected_, this design is _forwards
+compatible_:
+
+* Airflow metadata added today should remain usable.
+* **If** we ever add UI support for metadata from another (non-Airflow)
+  system, it should work with metadata added today.
+
+### Source of truth
+
+A [wise woman once wrote](https://twitter.com/mipsytipsy/status/998084191488126976):
+
+> You either have one source of truth, or multiple sources of lies.
+
+There is a natural tension and thereby a continuoum between these two
+extremes:
+
+* **Normalized data.**  lakeFS commits hold only enough to identify an
+  Airflow run - its URL.  Airflow holds all other run metadata, and is
+  required to make use of the URL.
+* **Denormalized data.**  lakeFS commits hold all the information that
+  lakeFS and its users need about an Airflow run.  All run metadata is
+  copied to lakeFS commit metadata.
+
+We always need the normalized data pointer, it is the only real identifier
+of the run.[^1] The lakeFS UI can offer the best integration when lakeFS is
+the source of truth.  Initially we shall allow users to select what metadata
+to copy.  The default will be _all_.  We may revisit this decision after
+feedback from users.
```

Comment:
I believe we should limit the scope even further. I think we should start with the leaner approach (i.e. just the url) and add capabilities based on users feedback, instead of removing capabilities which is always harder. IMO the MVP should be just the run url.

Reply:
Not sure.  To succeed it also needs to be valuable to users.  I want to put a bunch of metadata on commits, as long as it doesn't delay us.  I don't think any of this metadata is hard to get.

Note that the normalized data pointer is _not_ a link to the GUI for that DAG run, it's a link to the DAG run on the REST endpoint.

I'm adding a disclaimer that we'll step away from anything that's hard to get.  That gives us a better balance of velocity and usefulness.  Hope that's better!